### PR TITLE
Rename `SVGP` to `SparseVariationalApproximation`

### LINKED ISSUE #69: I'm in favour of renaming to `SparseVariationalApproximation`
I'm in favour of renaming to `SparseVariationalApproximation`

_Originally posted by @rossviljoen in https://github.com/JuliaGaussianProcesses/ApproximateGPs.jl/pull/68#issuecomment-961419453_

### PR DESCRIPTION
**Pull request by rossviljoen**
A true masterclass in find and replace

Closes #69

### CODE REVIEW

**Comment by rossviljoen** [comment]
Actually - should it be `SparseVariationalApproximation` or `SparseVariationalApproximation`? I suppose it isn't technically necessarily stochastic, but I thought the S in `SVGP` was Stochastic anyway?

**Comment by willtebbutt** [comment]
Good point. Truth be told, I'm not totally sure what people usually go with. @st-- @theogf @Crown421 do you have strong opinions?

**Comment by Crown421** [comment]
> Good point. Truth be told, I'm not totally sure what people usually go with. @st-- @theogf @Crown421 do you have strong opinions?

While the S in SVGP is "Stochastic", in the context of "Approximation" I would prefer "Sparse". "Stochastic Approximation" seems to have an existing connotation with root finding/ optimization.

**Comment by st--** [comment]
`SparseVariationalApproximation` please. As a side effect this allows you to get unbiased gradient estimates from mini-batches and hence stochastic optimisation, but that's not actually a requirement and the approximation doesn't depend on it being stochastic.

We could then also add a `VariationalApproximation` that makes it a bit easier/more convenient when you don't want to bother with inducing points.

(And I'm still very much looking forward to having an example of interdomain approximation:))

File: test/equivalences.jl [hunk 1]
```diff
@@ -11,7 +11,7 @@
     @testset "exact posterior" begin
         # There is a closed form optimal solution for the variational posterior
         # q(u) (e.g. # https://krasserm.github.io/2020/12/12/gaussian-processes-sparse/
-        # equations (11) & (12)). The SVGP posterior with this optimal q(u)
+        # equations (11) & (12)). The SparseVariationalApproximation posterior with this optimal q(u)
```
Comment (st--) [inline test/equivalences.jl:14]: This is one case where I would leave SVGP, as I think this is referring to the model (paper), not to the struct.
```suggestion
        # equations (11) & (12)). The SVGP posterior with this optimal q(u)
```

File: src/ApproximateGPs.jl [hunk 2]
```diff
@@ -15,10 +15,10 @@ using KLDivergences
 
 using AbstractGPs: AbstractGP, FiniteGP, LatentFiniteGP, ApproxPosteriorGP, At_A, diag_At_A
 
-export SVGP, DefaultQuadrature, Analytic, GaussHermite, MonteCarlo
+export SparseVariationalApproximation, DefaultQuadrature, Analytic, GaussHermite, MonteCarlo
```
Comment (st--) [inline src/ApproximateGPs.jl:18]: Can you add a `@deprecate SVGP SparseVariationalApproximation` somewhere (and we might have to keep the export here)?

File: examples/a-regression/script.jl [hunk 3]
```diff
@@ -123,9 +123,9 @@ function (m::SVGPModel)(x)
     return post(x)
 end;
 
-# Return the loss given data - for the SVGP, the loss used is the negative ELBO
-# (also known as the Variational Free Energy). `num_data` is required for
-# minibatching used below.
+# Return the loss given data - for the SparseVariationalApproximation, the
```
Comment (st--) [inline examples/a-regression/script.jl:126]:
```suggestion
# Return the loss given data - for the SVGP model as constructed using `SparseVariationalApproximation`, the
```

**Review by st-- — APPROVED** [review]
Just a few minor comments - happy for you to merge directly after you've addressed them:)

**Comment by st--** [comment]
NB- still needs a patch bump if you want to release it straight away

**Comment by rossviljoen** [comment]
The only thing I was worried about was that `VFE` could also be described as `SparseVariationalApproximation` - although perhaps it should be a special case of SVGP anyway...

Even with the deprecation, this is still breaking right? so should be 0.2? @st--

**Comment by theogf** [comment]
No if we deprecate `SVGP` it just means people will get a warning but it will still works, therefore it is non-breaking. However removing the `@deprecate` will make it breaking

**Comment by rossviljoen** [comment]
`@deprecate SVGP SparseVariationalApproximation` only provides an alias for the constructor, not the type (e.g. won't work if you're trying to dispatch on `SVGP`). So, have bumped the minor version, but keeping the `@deprecate`

**Pull request merged**